Improve ASA overview and add asset metadata page (#593)
algorandfoundation/devportal: src/content/docs/concepts/assets/overview.mdx

import { Code } from '@astrojs/starlight/components';
import { LinkCard } from '@astrojs/starlight/components';

> The Algorand protocol supports the creation of on-chain assets that benefit from the same security, compatibility, speed, and ease of use as the native ALGO asset. The official name for assets on Algorand is **Algorand Standard Assets (ASAs)**.

Reviewer: ```suggestion
The Algorand protocol supports the creation of on-chain assets that benefit from the same security, compatibility, speed, and ease of use as the native Algo asset. The official name for assets on Algorand is **Algorand Standard Assets (ASAs)**.
```

"Algo" is the correct usage. https://solas.algorand.foundation/doc/algorand-algorand-foundation-and-algo-HNqmBVbchV

> With Algorand Standard Assets you can represent stablecoins, loyalty points, system credits, and in-game points, among many other digital assets. You can also represent single, unique assets like a deed for a house, collectable items, and unique parts on a supply chain.

Reviewer: ```suggestion
With Algorand Standard Assets you can represent stablecoins, loyalty points, system credits, and in-game points, among many other digital assets. You can also represent single, unique assets like a deed for a house, collectible items, and unique parts on a supply chain.
```

| [AssetName](/concepts/transactions/reference#assetname)         | _No, but recommended_ | Full name of the asset, max 32 bytes. Example: `USD Coin`.              |
| [UnitName](/concepts/transactions/reference#unitname)           | _No, but recommended_ | Short ticker name, max 8 bytes. Example: `USDC`.                        |
| [Total](/concepts/transactions/reference#total)                 | _Yes_                 | Total number of base units to create. Cannot be changed after creation. |
| [Decimals](/concepts/transactions/reference#decimals)           | _Yes_                 | Digits after the decimal point when displaying the asset (0–19).        |
| [DefaultFrozen](/concepts/transactions/reference#defaultfrozen) | _Yes_                 | If `true`, asset holdings are frozen by default in every account.       |
| [URL](/concepts/transactions/reference#url)                     | _No_                  | URL with more information about the asset, max 96 bytes.                |
| [MetaDataHash](/concepts/transactions/reference#metadatahash)   | _No_                  | Optional 32-byte hash of asset-related metadata.                        |

### Mutable Asset Parameters

> Four address parameters control specific capabilities of an asset. These addresses are set during asset creation and determine who can perform actions such as updating configuration, freezing accounts, or clawing back tokens.

Reviewer: The sentence structure is a bit awkward and "address parameter" is a bit unclear.

I'd switch to something like: "There are four *configuration* parameters in an ASA, each store an address that..."

The four address types are:

<LinkCard
  title='Manager Address'
  href='/concepts/transactions/reference#manageraddr'
  description='Account that can update control addresses and destroy the asset'

> />

Reviewer: I like where you're going with the LinkCard component, but in practice it's very hard to read the content. Maybe do this as a list, and include the link card in the bottom of the section?

<img width="1284" height="909" alt="Image" src="https://github.com/user-attachments/assets/b270c076-1ed4-459a-a46a-e7cfb4591c7f" />

The **manager account** authorizes transactions that reconfigure or destroy an asset.

If a manager address is specified, it can later update the control addresses shown in this section.

<LinkCard
  title='Reserve Address'
  href='/concepts/transactions/reference#reserveaddr'
  description='Account that can hold non-minted asset units'
/>

> The reserve account holds **non-minted units** of the asset instead of the creator account.  

Reviewer: ```suggestion
The reserve account holds **non-circulating units** of the asset instead of the creator account.  
```
Technically all units are minted when the asset is created. Better to say "non-circulating" or "undistributed".

> Assets transferred from the reserve account are considered newly minted units.

Reviewer: There's no protocol-levle 
```suggestion
Assets transferred from the reserve account are considered units entering circulation.
```

Might note somewhere that the reserve address is an informational designation, not a protocol-enforced supply-control mechanism. Transfers from that address can be interpreted as distribution of reserve units, as most blockchain explorers do, but that interpretation is conventional rather than enforced by the protocol.
